src/main/java/io/sqreen/powerwaf/Additive.java

```diff
@@ -83,9 +85,15 @@ public Powerwaf.ResultWithData run(Map<String, Object> parameters,
                     if (!online) {
                         throw new IllegalStateException("This Additive is no longer online");
                     }
-                    ByteBuffer bb;
+                    ByteBuffer persistentBuffer = null;
+                    ByteBuffer ephemeralBuffer = null;
                     try {
-                        bb = this.lease.serializeMore(limits, parameters);
+                        if (persistentData != null) {
+                            persistentBuffer = this.lease.serializeMore(limits, persistentData);
+                        }
+                        if (ephemeralData != null) {
+                            ephemeralBuffer = this.lease.serializeMore(limits, ephemeralData);
+                        }
```

Comment:
The ephemeral data should use a separate lease that we return immediately after the call. The reason is that otherwise the memory can grow up quite a lot, as the ephemeral data is made with the objective of sending data for the same address repeatedly.

Reply:
Done